stan/math/prim/prob/beta_neg_binomial_lccdf.hpp

```diff
@@ -0,0 +1,151 @@
+ * @tparam T_n type of failure parameter
+ * @tparam T_r type of number of successes parameter
+ * @tparam T_alpha type of prior success parameter
+ * @tparam T_beta type of prior failure parameter
+ *
+ * @param n failure parameter
+ * @param r Number of successes parameter
+ * @param alpha prior success parameter
+ * @param beta prior failure parameter
+ * @return log probability or log sum of probabilities
+ * @throw std::domain_error if r, alpha, or beta fails to be positive
+ * @throw std::invalid_argument if container sizes mismatch
+ */
+template <typename T_n, typename T_r, typename T_alpha, typename T_beta>
+inline return_type_t<T_r, T_alpha, T_beta> beta_neg_binomial_lccdf(
+    const T_n& n, const T_r& r, const T_alpha& alpha, const T_beta& beta) {
+  using std::exp;
+  using std::log;
+  using T_partials_return = partials_return_t<T_n, T_r, T_alpha, T_beta>;
+  using T_r_ref = ref_type_t<T_r>;
+  using T_alpha_ref = ref_type_t<T_alpha>;
+  using T_beta_ref = ref_type_t<T_beta>;
+  static constexpr const char* function = "beta_neg_binomial_lccdf";
+  check_consistent_sizes(
+      function, "Failures variable", n, "Number of successes parameter", r,
+      "Prior success parameter", alpha, "Prior failure parameter", beta);
+  if (size_zero(n, r, alpha, beta)) {
+    return 0;
+  }
+
+  T_r_ref r_ref = r;
+  T_alpha_ref alpha_ref = alpha;
+  T_beta_ref beta_ref = beta;
+  check_positive_finite(function, "Number of successes parameter", r_ref);
+  check_positive_finite(function, "Prior success parameter", alpha_ref);
+  check_positive_finite(function, "Prior failure parameter", beta_ref);
+
+  T_partials_return P(0.0);
+  auto ops_partials = make_partials_propagator(r_ref, alpha_ref, beta_ref);
+
+  scalar_seq_view<T_n> n_vec(n);
+  scalar_seq_view<T_r_ref> r_vec(r_ref);
+  scalar_seq_view<T_alpha_ref> alpha_vec(alpha_ref);
+  scalar_seq_view<T_beta_ref> beta_vec(beta_ref);
+  size_t size_n = stan::math::size(n);
+  size_t max_size_seq_view = max_size(n, r, alpha, beta);
+
+  // Explicit return for extreme values
+  // The gradients are technically ill-defined, but treated as zero
+  for (size_t i = 0; i < size_n; i++) {
+    if (n_vec.val(i) < 0) {
+      return ops_partials.build(0.0);
+    }
+  }
+
+  for (size_t i = 0; i < max_size_seq_view; i++) {
+    // Explicit return for extreme values
+    // The gradients are technically ill-defined, but treated as zero
+    if (n_vec.val(i) == std::numeric_limits<int>::max()) {
+      return ops_partials.build(negative_infinity());
+    }
+    T_partials_return n_dbl = n_vec.val(i);
+    T_partials_return r_dbl = r_vec.val(i);
+    T_partials_return alpha_dbl = alpha_vec.val(i);
+    T_partials_return beta_dbl = beta_vec.val(i);
+    T_partials_return b_plus_n = beta_dbl + n_dbl;
+    T_partials_return r_plus_n = r_dbl + n_dbl;
+    T_partials_return a_plus_r = alpha_dbl + r_dbl;
+    T_partials_return one = 1;
+    T_partials_return precision = 1e-8;  // default -6, set -8 to pass all tests
```

Comment:
Yes, should I also expose the number of max iteration? Because if user use high precise, it may exceeds max iteration and throw error, refering to #2158.

Reply:
Yes I think that makes sense